docs/configuration.md

```diff
@@ -22,4 +22,71 @@ For the complete list of settings GCM Core understands, see the list below.
 
 ## Available settings
 
-_Currently there are no available settings. This is a placeholder for the settings on the backlog._
+### credential.provider
+
+Define the host provider to use when authenticating.
+
+ID|Provider
+-|-
+`auto` _(default)_|_\[automatic\]_
+`azure-repos`|Azure Repos
+`github`|GitHub
+`generic`|Generic
+
```

Comment:
You may want to add a description of what we mean by Generic.  It can be something as simple as "If you are using a provider other than Azure Repos or Github, choice Generic"

Reply:
Added.